yearn/apy/curve/simple.py

```diff
@@ -81,12 +85,35 @@ def simple(vault, samples: ApySamples) -> Apy:
     if vault.vault.address == "0x46AFc2dfBd1ea0c0760CAD8262A5838e803A37e5":
         boost = 1
 
-
+    # TODO: come up with cleaner way to deal with these new gauge rewards
     reward_apr = 0
     if hasattr(gauge, "reward_contract"):
         reward_address = gauge.reward_contract()
         if reward_address != ZERO_ADDRESS:
             reward_apr = rewards(reward_address, pool_price, base_asset_price, block=block)
+    elif hasattr(gauge, "reward_data"): # this is how new gauges, starting with MIM, show rewards
+        # get our token
+        gauge_reward_token = gauge.reward_tokens(0)
+        if gauge_reward_token == RKP3R_REWARDS or gauge_reward_token == NULL_ADDRESS:
+            print("\nrKP3R gauge or no reward token")
+        else:
+            # get our total supply
+            _distributor = gauge.reward_data(gauge_reward_token)[1]
+            token_contract = Contract(gauge_reward_token)
+            total_supply = gauge.totalSupply()
+            
+            # get our rate
+            rate = gauge.reward_data(gauge_reward_token)[3]
+            
+            token_price = get_price(gauge_reward_token, block=block)
+            
+            # get our period end
+            period_finish = gauge.reward_data(gauge_reward_token)[2]
```

Comment:
Brownie allows accessing struct return values by field names, it looks a bit clearer than indexes. Also, there is no need to query the node multiple times for the same thing.
```suggestion
            reward_data = gauge.reward_data(gauge_reward_token)
            rate = reward_data['rate']
            period_finish = reward_data['period_finish']
            token_contract = Contract(gauge_reward_token)
            total_supply = gauge.totalSupply()
            token_price = get_price(gauge_reward_token, block=block)
```